File: Greedy/#1202.py
import sys
import heapq

N, K = map(int, sys.stdin.readline().split())
jewelry = []
for _ in range(N):
    heapq.heappush(jewelry, list(map(int, sys.stdin.readline().split())))
bags = []
for _ in range(K):
    bags.append(int(sys.stdin.readline()))
bags.sort()

answer = 0
tmp_jew = []
for bag in bags:
    while jewelry and bag >= jewelry[0][0]:
        heapq.heappush(tmp_jew, -heapq.heappop(jewelry)[1])
    if tmp_jew:
        answer -= heapq.heappop(tmp_jew)
    elif not jewelry:
        break
print(answer)

File: Greedy/#1202.js
const _input = require("fs").readFileSync("sample.txt").toString().trim().split("\n");

function solution(input) {
    let answer = 0;
    const jNum = +input[0].split(" ")[0];
    const bNum = +input[0].split(" ")[1];

    const jewelry = input
        .slice(1, input.length - bNum)
        .map((item) => {
            return [+item.split(" ")[0], +item.split(" ")[1]];
        })
        .sort((a, b) => {
            return b[1] - a[1];
        });

    const bag = input
        .slice(1 + jNum)
        .map((item) => {
            return +item;
        })
        .sort((a, b) => {
            return b - a;
        });

    for (let i = 0; i < bNum; i++) {
        let j = 0;
        while(1){
            if (jewelry[j][0]!== -1 && bag[i] >= jewelry[j][0]) {
                answer += jewelry[j][1];
                jewelry[j][0] = -1;
                break;
            }
            j++;
        }
    }
    return answer;
}

console.log(solution(_input));
